tests/cases/fourslash/refactorConvertStringOrTemplateLiteral_escapeSequences.ts

```diff
@@ -0,0 +1,45 @@
+/// <reference path='fourslash.ts' />
+
+// @Filename: /a.ts
+////let s = /*a1*/"\0\b\f\t\r\n" + text + "\n"/*a2*/;
+
+goTo.select("a1", "a2");
+edit.applyRefactor({
+    refactorName: "Convert to template string",
+    actionName: "Convert to template string",
+    actionDescription: ts.Diagnostics.Convert_to_template_string.message,
+    newContent: 'let s = `\\u0000\\b\\f\\t\\r\\n${text}\\n`;'
+});
+
+// @Filename: /b.ts
+////let s = /*b1*/'"' + text + "'"/*b2*/;
+
+goTo.select("b1", "b2");
+edit.applyRefactor({
+    refactorName: "Convert to template string",
+    actionName: "Convert to template string",
+    actionDescription: ts.Diagnostics.Convert_to_template_string.message,
+    newContent: 'let s = `"${text}\'`;'
+});
+
+// @Filename: /c.ts
+////let s = /*c1*/'${}' + text + "\\"/*c2*/;
+
+goTo.select("c1", "c2");
+edit.applyRefactor({
+    refactorName: "Convert to template string",
+    actionName: "Convert to template string",
+    actionDescription: ts.Diagnostics.Convert_to_template_string.message,
+    newContent: 'let s = `\\${}${text}\\\\`;'
```

Comment:
This is an incorrect result (it’s invalid syntax). It should be

```suggestion
    newContent: 'let s = `\${}${text}\\\\`;'
```

Reply:
@andrewbranch: I think that the quotes here are confusing: that `\\` is needed since the whole thing is in a ``'...`...`...'``, so it's actually `'`-quoted which needs the `\\`.  And to make it more confusing, that output is for the fourslash input which is quoted differently...  Two ways to see it more mechanically:

* Removing that `\` will leave a string that has a bad escape.

* Another way is to evaluate the code:

  ```ts
  > let text = "text";
  undefined
  > let s1 = /*c1*/'${}' + text + "\\"/*c2*/;     // the original form from the fourslash test w/ s1
  undefined
  > console.log('let s = `\\${}${text}\\\\`;');   // the string from this line, copied as-is
  let s = `\${}${text}\\`;
  > let s2 = `\${}${text}\\`;                     // copy it w/ s2
  undefined
  > s1 === s2
  true
  ```

Reply:
🤦 of course. I’ve written my own fair share of double-escaped tests but totally read this one literally. My bad!